services/graph/pkg/identity/ldap.go

```diff
@@ -730,3 +740,19 @@ func (i *LDAP) expandLDAPAttributeEntries(ctx context.Context, e *ldap.Entry, at
 
 	return result, nil
 }
+
+func replaceDN(fullDN *ldap.DN, newDN string) (string, error) {
+	if len(fullDN.RDNs) == 0 {
+		return "", fmt.Errorf("Can't operate on an empty dn")
+	}
+
+	if len(fullDN.RDNs) == 1 {
+		return newDN, nil
+	}
+
+	for _, part := range fullDN.RDNs[1:] {
+		newDN += "," + part.String()
+	}
```

Comment:
How about:

```
baseDN := &DN{
   RDN: fullDN.RDNs[1:]
}
newDN += baseDN.String()
```

Seems to be a little cleaner and might be a bit more efficient.

Reply:
You're right, was a bit in tunnel vision mode. :)